packages/react/src/lib/factories.ts

```diff
@@ -91,6 +92,10 @@ type CreateShorthandFactoryConfigInner<TPropName = string> = {
   mappedArrayProp?: TPropName
 }
 export type CreateShorthandFactoryConfig = CreateShorthandFactoryConfigInner
+export type CreateShorthandFactoryResult = (
+  a: ShorthandValue<UIComponentProps>,
```

Comment:
```suggestion
  value: ShorthandValue<UIComponentProps>,
```

Reply:
```suggestion
  a: ShorthandValue<P>,
```

Actually, let's make it generic and accept component props there. Hope that it will work

Reply:
Yes it works